packages/forms/src/utils/renderComponent.js

```diff
@@ -0,0 +1,28 @@
+// @flow
+import * as React from 'react';
+
+import type { RenderableProps } from '../types';
+
+const renderComponent = (props: RenderableProps) => {
+  const { render, children, component, ...rest } = props;
+
+  let rendered = null;
+
+  if (component) {
+    rendered = React.createElement(component, { ...rest }, children);
+  }
+
+  if (render) {
+    rendered = render({ ...rest, children });
+  }
+
+  if (typeof children === 'function') {
+    rendered = children(rest);
+  } else {
+    rendered = children;
+  }
+
+  return rendered;
+};
```

Comment:
What do you think about adding warnings in cases when user passed `component` prop with `children` or `render` props.

Reply:
@ej9x good idea